engine/chain/internal/bhs/get_merkleroots_from_bhs.go

```diff
@@ -0,0 +1,64 @@
+package bhs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"syscall"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+)
+
+// GetMerkleRootsFromBHS returns Merkle Roots from Block Header Service
+func (s *Service) GetMerkleRootsFromBHS(ctx context.Context, query url.Values) (*models.MerkleRootsBHSResponse, error) {
+	bhsURL, err := s.createBHSURL("/chain/merkleroot")
+	if err != nil {
+		return nil, err
+	}
+
+	req := s.httpClient.R().
+		SetContext(ctx).
+		SetHeader("Content-Type", "application/json").
+		EnableTrace()
+
+	if s.bhsCfg.AuthToken != "" {
+		req.SetAuthToken(s.bhsCfg.AuthToken)
+	} else {
+		s.logger.Warn().Msg("warning creating Block Headers Service url - auth token is not set. Some requests might not work")
+	}
+
+	var response models.MerkleRootsBHSResponse
+	res, err := req.
+		SetResult(&response).
+		SetQueryParamsFromValues(query).
+		Get(bhsURL.String())
+	if err != nil {
+		if errors.Is(err, syscall.ECONNREFUSED) {
+			return nil, chainerrors.ErrBHSUnreachable.Wrap(err)
+		}
+		return nil, spverrors.ErrInternal.Wrap(err)
+	}
+	if !res.IsSuccess() {
+		return nil, mapBHSErrorResponseToSpverror(res)
+	}
+
+	return &response, nil
+}
+
+// createBHSURL parses Block Header Url from configuration and constructs a valid
+// endpoint with provided endpointPath variable
+func (s *Service) createBHSURL(endpointPath string) (*url.URL, error) {
+	if s.bhsCfg.URL == "" {
+		s.logger.Error().Msgf("create Block Header Service URL - url not configured")
+	}
```

Comment:
I didn't check the URL at this level. And I'm not convinced we have to do so 🤔 
Besides, we do the verification at `config` level:
https://github.com/bitcoin-sv/spv-wallet/blob/ce45722045af9f82c57d4d8935492b52d4dd0e17/config/validate_arc.go#L21-L23

Reply:
mm yeah, I copied what I created in a previous version but since we are checking it at the config level there is no need to do it twice, removed